pkg/cli/admin/nodeimage/create.go

```diff
@@ -116,17 +138,36 @@ type CreateOptions struct {
 	AssetsDir string
 	// OutputName allows the user to specify the name of the generated image.
 	OutputName string
+	// Simpler interface for creating a single node
+	SingleNodeOpts *singleNodeCreateOptions
 
 	nodeJoinerExitCode int
 	rsyncRshCmd        string
 }
 
+type singleNodeCreateOptions struct {
+	MacAddress      string
+	CPUArchitecture string
+	SSHKey          string
+	Hostname        string
+	RootDeviceHints string
+	NetworkConfig   string
+}
+
 // AddFlags defined the required command flags.
 func (o *CreateOptions) AddFlags(cmd *cobra.Command) {
 	flags := o.addBaseFlags(cmd)
 
 	flags.StringVar(&o.AssetsDir, "dir", o.AssetsDir, "The path containing the configuration file, used also to store the generated artifacts.")
 	flags.StringVarP(&o.OutputName, "output-name", "o", "node.iso", "The name of the output image.")
+
+	flags.StringP(snFlagMacAddress, "m", "", "Single node flag. MAC address used to identify the host to apply the configuration. If specified, the nodes-config.yaml config file will not be used.")
+	usageFmt := "Single node flag. %s. Valid only when `mac-address` is defined."
+	flags.StringP(snFlagCpuArch, "c", "", fmt.Sprintf(usageFmt, "The CPU architecture to be use to install the node"))
+	flags.StringP(snFlagSshKey, "k", "", fmt.Sprintf(usageFmt, "The SSH key used to access the node"))
+	flags.String(snFlagHostname, "", fmt.Sprintf(usageFmt, "The hostname to be set for the node"))
+	flags.String(snFlagRootDeviceHint, "", fmt.Sprintf(usageFmt, "Hint for specifying the storage location for the image root filesystem"))
```

Comment:
nit: `Root device hint to specify the storage location for the image root filesystem` Similar verbiage to other flags.

Reply:
It would be helpful to mention what is the root device hint format. Can the help refer to a url that describes the format?

Reply:
I though to delegate more of this stuff to the documentation, but probably a direct mention here could be useful